bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -108,7 +108,7 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
         logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
+        if (!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {
```

Comment:
@kaikreuzer, 

postUpdate behaves identically in OH 1 and 2.

Differences in log new installed OH 1.8.3 and OH2b4:
### OH1 sendCommand ON/OFF
```
2016-10-24 14:21:45.257 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='ON')
2016-10-24 14:21:45.261 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-24 14:21:45.263 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-24 14:21:45.263 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Wrote value 'ON' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-24 14:21:45.264 [WARN ] [.b.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.243 for Group address 1/1/10.

2016-10-24 14:21:50.017 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-24 14:21:50.022 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-24 14:21:50.024 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Wrote value 'OFF' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-24 14:21:50.024 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-24 14:21:50.026 [WARN ] [.b.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.243 for Group address 1/1/10.
```
### OH2 sendCommand ON/OFF
```
2016-10-24 15:16:20.005 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-24 15:16:20.005 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='OFF')
2016-10-24 15:16:20.005 [TRACE] [g.knx.internal.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-24 15:16:20.006 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-24 15:16:20.006 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Wrote value 'OFF' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-24 15:16:20.006 [WARN ] [.binding.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.244 for Group address 1/1/10.

2016-10-24 15:16:25.005 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='ON')
2016-10-24 15:16:25.005 [TRACE] [g.knx.internal.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-24 15:16:25.005 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='ON')
2016-10-24 15:16:25.006 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Wrote value 'ON' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-24 15:16:25.006 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-24 15:16:25.006 [WARN ] [.binding.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.244 for Group address 1/1/10.
```
My Item and Rules;
```
Switch TestSwitch 	 		"TestSwitch" 		 ()	 { knx="<1/1/10" }
```
```
import org.openhab.core.library.types.*
import org.openhab.core.persistence.*
import org.openhab.model.script.actions.*
/*
rule "knxTest_postUpdate"
	when
		Time cron "0/5 * * * * ?"
	then
	if(TestSwitch.state == OFF) {
		TestSwitch.postUpdate( ON )
	} else {
		TestSwitch.postUpdate( OFF )
	}	
end
*/
/*
rule "knxTest_sendCommand"
	when
		Time cron "0/5 * * * * ?"
	then
	if(TestSwitch.state == OFF) {
		TestSwitch.sendCommand( ON )
	} else {
		TestSwitch.sendCommand( OFF )
	}
end
*/
```

OH2 is much more faster! :-)
And for me OH2s behavior is correct.

The OH1 behavior is very old, because if not, we where stumbled over loosing/repeated telegrams as since compat1x layer in OH2.

@watou, 
I must not flip  `(!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {`, it deletes the echo marker and Echo will be sent to KNX when called `internalReceiveUpdate` after `internalReceiveCommand`.

I stressed the KNX-bus and KNX1 binding with hundreds of fast changed calls. The ignoreEventList never got longer as 11 records and all these records where removed cleanly.

But to prevent ignoreEventList to grow unstopped and leak, I now use Set<String> instead of List<String>. now It cannot contain duplicate elements and is thus limited in size. 

When we find the way how OH1 and compat1x layer in OH2 works identically we do not need the compatibilityOH2 switch. I hope you two will help me, I miss your deep insights into the classes in background!